luxer_invoice_update/models/sale_order.py

```diff
@@ -0,0 +1,19 @@
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    site_address = fields.Many2one('res.partner', string='Site Address')
+
+    def action_invoice_subscription(self):
+        if self.site_address:
+            account_move = self._create_recurring_invoice()
```

Comment:
since you need the record set created from _create_recurring_invoice, you could just inherit that method and add your own logic after that super call instead of here, but this probably works too
will be less disruptive overall